devito/operator/operator.py

```diff
@@ -856,7 +857,13 @@ def apply(self, **kwargs):
         # Post-process runtime arguments
         self._postprocess_arguments(args, **kwargs)
 
-        # Output summary of performance achieved
+        # In case MPI is used restrict result logging to one rank only
+        if configuration['mpi']:
+            # Only temporarily change configuration
+            with switchconfig(mpi=True):
+                set_log_level('DEBUG', comm=args.comm)
```

Comment:
Again: you're switching to DEBUG and then it remains DEBUG once control is handed back to the user

it shouldn't be DEBUG either, it should be whatever was set up to this point

and finally we shouldn't switch to mpi=True (why btw?) because it will trigger a compiler rebuild which we should (can) avoid: https://github.com/devitocodes/devito/blob/master/devito/__init__.py#L74

Reply:
Oh right. Makes sense. Thanks